components/Volunteer/FutureVolunteerEvents.tsx

```diff
@@ -69,12 +82,17 @@ const FutureVolunteerEvents = () => {
     <>
       {/* TODO: render 404 page */}
       {error && <>404 Event not found!</>}
-      {!events && !error && <div>Loading...</div>}
-      {events && (
+      {(!events || !featuredEvents) && !error && <div>Loading...</div>}
```

Comment:
DeMorgan's principle 
```suggestion
      {!(events && featuredEvents) && !error && <div>Loading...</div>}
```